main.py

```diff
@@ -30,74 +41,128 @@
+    :type duration: int
+    :raises ValueError: if duration is zero or negative
+    """
+    def outer_wrapper(func):
+        @functools.wraps(func)
+        def inner_wrapper(*args, **kwargs):
+            current_time = datetime.now()
+            key = hashlib.md5((func.__name__ + str(args) + str(kwargs)).encode("utf-8")).hexdigest()
+            fname = os.path.join(INFO_CACHE, key+".pcl")
+            try:
+                if not xbmcvfs.exists(fname):
+                    raise KeyError
+                xbmc.log(f'[ByteFM] Exists: {fname}', xbmc.LOGDEBUG)
+                timestamp = datetime.fromtimestamp(xbmcvfs.Stat(fname).st_mtime())
+                xbmc.log(f'[ByteFM] Timestamp: {timestamp}', xbmc.LOGDEBUG)
+                if current_time - timestamp > timedelta(minutes=duration):
+                    raise KeyError
+                xbmc.log(f'[ByteFM] Read: {fname}', xbmc.LOGDEBUG)
+                with open(fname, "rb") as fp:
+                    data = pickle.load(fp)
+                xbmc.log(f'[ByteFM] Cache hit: {key}', xbmc.LOGDEBUG)
+            except (KeyError, pickle.UnpicklingError):
+                xbmc.log(f'[ByteFM] Cache miss: {key}', xbmc.LOGDEBUG)
+                data = func(*args, **kwargs)
+                with open(fname, "wb") as fp:
+                    pickle.dump(data, fp)
+            return data
+        return inner_wrapper
+    return outer_wrapper
+
+
+def plugin_url(**kwargs):
+    """Build plugin:// url with args."""
+    if kwargs:
+        return sys.argv[0] + '?' + urlencode(kwargs, doseq=True)
+    else:
+        return sys.argv[0]
+
+
+session = requests.session()
+session.headers['User-Agent'] = xbmc.getUserAgent()
+try:
+    session.auth = (
+        ADDON.getSettingString("byte.login.username"),
+        ADDON.getSettingString("byte.login.password")
+    )
+except Exception:
+    xbmc.log("[ByteFM] Credentials not set. Using None.", xbmc.LOGERROR)
+
 def _http_get(url, **kwargs):
     """Log and perform a HTTP GET"""
-    plugin.log_notice("HTTP GET: {}".format(url))
-    kwargs['auth'] = AUTH
-    resp = requests.get(url, **kwargs)
+    xbmc.log(f"[ByteFM] HTTP GET: {url}", xbmc.LOGINFO)
+    resp = session.get(url, **kwargs)
     try:
         resp.raise_for_status()
     except HTTPError as e:
-        if e.response.status_code == 401:
-            xbmcgui.Dialog().ok(
-                'ByteFM', _("Authentication Failed!"),
-                _("Please check your username and password."), '')
-            plugin.addon.openSettings()
+        if e.response is not None and e.response.status_code == 401:
+            xbmcgui.Dialog().ok('ByteFM', __localize__(32008),  __localize__(32009), '')
+            ADDON.openSettings()
             sys.exit(-1)
         else:
             raise
     return resp
 
 def _strip_html(text):
-    text = text or ''
-    return BeautifulSoup(text).getText()
+    if not text:
+        return ''
+    # Remove all tags
+    return re.sub(r'<[^>]+>', '', text)
```

Comment:
Not sure if this is working correctly, see Kaleidoscop:

<img width="1112" height="594" alt="Screenshot 2025-08-17 at 19 21 47" src="https://github.com/user-attachments/assets/ccba67e6-65df-4158-8194-2b9d6db9ea62" />

(or perhaps `_strip_html()` is not being called in the appropriate places)

Reply:
This place *should* work:
```Python
>>> import re
>>> text = '<p><span style="color:black;">Nach den Ferien ist vor den…</span></p>'
>>> re.sub(r'<[^>]+>', '', text)
'Nach den Ferien ist vor den…'
```
but I will check more carefully.